Tests/UtilityTests/ProcessTests.swift

```diff
@@ -0,0 +1,92 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import XCTest
+import Basic
+import libc
+import Utility
+
+class ProcessTests: XCTestCase {
+
+    func testBasics() throws {
+        let process = Process(args: "echo", "hello")
+        try process.launch()
+        let result = try process.waitUntilExit()
+        XCTAssertEqual(try result.output.dematerialize(), "hello\n")
+        XCTAssertEqual(result.exitStatus, .terminated(code: 0))
+    }
+
+    func testPopen() throws {
+        XCTAssertEqual(try Process.popen(arguments: ["echo", "hello"]).output.dematerialize(), "hello\n")
+
+        let path = AbsolutePath(#file).parentDirectory.appending(component: "ArgumentParserTests.swift")
+        XCTAssertGreaterThan(try Process.popen(args: "cat", path.asString).output.dematerialize().characters.count, 4096)
+
+        let result = try Process.popen(args: "cat", "/bin/cat")
+        // The process should have succeeded.
+        XCTAssertEqual(result.exitStatus, .terminated(code: 0))
+        do {
+            _ = try result.output.dematerialize()
+            XCTFail("reading succeeded but should have failed")
+        } catch {}
+    }
+
+    func testSignals() throws {
+        func script(_ name: String) -> String {
+            return AbsolutePath(#file).parentDirectory.appending(components: "processInputs", name).asString
+        }
+        func running(_ pid: String) throws -> Bool {
+            let p = try Process.popen(args: "ps", "-p", pid)
+            // This is a hack for when running just tests via docker.
+            if try p.output.dematerialize().contains("defunct") {
+                return false
+            }
+            return p.exitStatus == .terminated(code: 0)
+        }
+
+        // Test sigint terminates the script.
+        do {
+            let process = Process(args: script("print-pid"))
+            try process.launch()
+            usleep(10000) // Need some sleep to allow the process to launch.
+            // Ensure process has started running.
+            XCTAssertTrue(try running("\(process.processID)"))
+            process.signal(.interrupt)
+            let result = try process.waitUntilExit()
+            XCTAssertEqual(result.exitStatus, .signalled(signal: Signal.interrupt.rawValue))
+            XCTAssertEqual("\(process.processID)\n", try result.output.dematerialize())
+            XCTAssertFalse(try running("\(process.processID)"))
+        }
+
+        // Test SIGKILL terminates the subprocess and any of its subprocess.
+        do {
+            let process = Process(args: script("subprocess"))
```

Comment:
I think this test would be easier to read if the subtask wrote a file containing its pid and its subprocess PID. You could just manually write a tidy `JSON` file using `printf(1)`